coderd/database/dbauthz/dbauthz.go

```diff
@@ -3906,6 +3920,13 @@ func (q *querier) UpsertProvisionerDaemon(ctx context.Context, arg database.Upse
 	return q.db.UpsertProvisionerDaemon(ctx, arg)
 }
 
+func (q *querier) UpsertRuntimeConfig(ctx context.Context, arg database.UpsertRuntimeConfigParams) error {
+	if err := q.authorizeContext(ctx, policy.ActionCreate, rbac.ResourceSystem); err != nil {
```

Comment:
Does the usage of `rbac.ResourceSystem` here mean that only site-wide owners can edit runtime config?

Reply:
I intentionally am making this vague atm. 

Once we start adding APIs to interact with the configs, we will probably have to write custom authorize calls depending on the setting. 

The goal of this layer is to be able to add a consistent caching layer infront, so we don't need to implement some pubsub notify go routine for each setting. When we implement the caching layer, it will skip calling the `dbauthz` layer entirely on cache hits.

So the RBAC story has to be solved for this layer, and I'm kicking the can on it for now. This is the core, basic CRUD.

Reply:
Here is an example where we skip crud on cached values today:

https://github.com/coder/coder/blob/e461d2773b5a2f1bd04ff899a922032b373e187f/agent/agent.go#L713-L740

So this PR does not address it, but it is not addressed in current day runtime configs

Reply:
Can we add a TODO/FIXME here, with an issue link?

Reply:
I don't think we need to since it requires system auth right now, which is the highest.

So any user of this package essentially needs to solve it on their own until a general solution is created. (I might whip up something).

So this defaults to "secure" essentially.

Reply:
If we're kicking the work down the road, it's definitely good idea to create an issue to track it. Doesn't have to be in the `coder/coder` repo.

Reply:
Added issue: https://github.com/coder/coder/issues/14587